lucene/misc/src/java/org/apache/lucene/misc/index/IndexSplitter.java

```diff
@@ -123,8 +118,8 @@ private SegmentCommitInfo getInfo(String name) {
 
   public void remove(String[] segs) throws IOException {
     for (String n : segs) {
-      int idx = getIdx(n);
-      infos.remove(idx);
+      SegmentCommitInfo info = getInfo(n);
+      infos.remove(info);
```

Comment:
SegmentInfos.remove(int) isn't accessible from external packages. This method call can be switched to SegmentInfo.remove(SegmentCommitInfo); that is already opened to public.